docs/developer-docs/smart-contracts/compile.mdx

```diff
@@ -0,0 +1,75 @@
+---
+keywords: [beginner, tutorial, compile]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+import { GlossaryTooltip } from "/src/components/Tooltip/GlossaryTooltip";
+
+# Compile
+
+<MarkdownChipRow labels={["Beginner", "Tutorial"]} />
+
+## Overview
+
+After you have written the code for your project's <GlossaryTooltip>canisters</GlossaryTooltip>, you need to compile the code into a WebAssembly module before it can be deployed on ICP. 
+
+## How code is compiled to Wasm
+
+WebAssembly (Wasm) is platform independent and designed to run on a stack-based virtual machine. Wasm itself is not a programming language; it is the result of a programming language being comnpiled and converted into executable machine code. When canister code is compiled into Wasm, the system compiler creates a Wasm module in either a binary format (.wasm) or a compressed binary (.wasm.gz).
+
+This generated Wasm module includes a runtime system which implements a garbage collector function and uses the Wasm heap storage. Then, this module is implemented when the canister's code is installed. 
```

Comment:
>  a runtime system which implements a garbage collector function 

The presence of the garbage collector depends on the language. For example Rust doesn't have a garbage collection.

> Then, this module is implemented when the canister's code is installed. 

I don't understand this sentence. Usually, developers implement code and the compile it to Wasm binary/module, so the module already exists regardless of canister code installation.